starter/script.js

```diff
@@ -1 +1,136 @@
+  switch(arrayHolder[0]) {
+    case 0:
+      friend = window.prompt(arrayHolder[1])
+      currentPrompt = arrayHolder[3]
+      break;
+    case 1:
+      window.alert(arrayHolder[1]);
+      currentPrompt = arrayHolder[2]
+      break;
+    case 2:
+      switch(answerHold = window.prompt(arrayHolder[1])){
+        case arrayHolder[2]:
+          arrayHolder[9]= checkAnswerType(currentPrompt,answerHold);
+          currentPrompt = arrayHolder[3]
+          break;
+        case arrayHolder[4]:
+          arrayHolder[9]= checkAnswerType(currentPrompt,answerHold);
+          currentPrompt = arrayHolder[5]
+          break;
+        default:
+          currentPrompt = currentPrompt
+          break;
+      }
+      break;
+    case 3:
+      switch(answerHold = window.prompt(arrayHolder[1])){
+        case arrayHolder[2]:
+          arrayHolder[9]= checkAnswerType(currentPrompt,answerHold);
+          currentPrompt = arrayHolder[3]
+          break;
+        case arrayHolder[4]:
+          arrayHolder[9]= checkAnswerType(currentPrompt,answerHold);
+          currentPrompt = arrayHolder[5]
+          break;
+        case arrayHolder[6]:
+          arrayHolder[9]= checkAnswerType(currentPrompt,answerHold);
+          currentPrompt = arrayHolder[7]
+          break;
+        default:
+          currentPrompt = currentPrompt
+          break;
+      }
+      break;
+    }
+}
+while (arrayHolder[9] !=1);
+window.alert(endingMap[currentPrompt])
+window.alert('Thank you for playing!')
+
+function ageQuery() {
+  ageHolder = window.prompt("So erm... How old are you again?");
+  console.log(ageHolder)
+  if (131 < ageHolder) {
+    currentPromptHolder = "prompt02";
+  } else if (ageHolder == 131) {
+    currentPromptHolder = "prompt03"
+  } else if (111 < ageHolder && ageHolder < 131) {
+    currentPromptHolder = "prompt04";
+  } else if (ageHolder == 111) {
+    currentPromptHolder = "prompt05";
+  } else if (50 < ageHolder && ageHolder < 111) {
+    currentPromptHolder = "prompt06";
+  } else if (ageHolder == 50) {
+    currentPromptHolder = "prompt07";
+  } else if (0 < ageHolder && ageHolder< 50) {
+    currentPromptHolder = "prompt09";
+  } else {
+    window.alert("Oh come now, let\'s try this again...");
+    currentPromptHolder = ageQuery();
+  }
+  console.log(currentPromptHolder)
+  console.log(ageHolder)
+  return currentPromptHolder;
+}
+function checkAnswerType(prompt,answer){
+  if (prompt=="prompt02" && answer == "Y") {
+    return 1
+  } else if (prompt=="prompt03" && answer == "Y"){
+    return 1
+  } else if (prompt=="prompt03" && answer == "N"){
+    return 1
+  } else if (prompt=="prompt04" && answer == "S"){
+    return 1
+  } else if (prompt=="prompt05" && answer == "F"){
+    return 1
+  } else if (prompt=="prompt05" && answer == "D"){
+    return 1
+  } else if (prompt=="prompt08" && answer == "N"){
+    return 1
+  } else if (prompt=="prompt09" && answer == "B"){
+    return 1
+  } else {
+    return 0
+  }
+}
```

Comment:
I like the thought behind your program architecture. I think it could really be simplified and made more elegant if you separate/decouple the map of the flow of events from the questions (what you have in `promptMap`). It's good to keep your data-structures well-organized and as simple as possible. Using `true` and `false` is preferable to using 1 and 0 as well, since they are more semantic. 

The robust effort and thought behind this assignment is very admirable and as is your willingness to tack complexity.